src/api/controllers/BaseController.ts

```diff
@@ -0,0 +1,40 @@
+import { EventEmitter } from 'eventemitter3';
+
+export interface BaseEvents
+{
+    hidden: ()=> void;
+    shown: ()=> void;
+    opened: ()=> void;
+    closed: ()=> void;
+}
+
+export interface BaseControllerOptions
+{
+    style?: any;
+    view?: any;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+export abstract class BaseController<EVENTS extends BaseEvents = BaseEvents> extends EventEmitter<EVENTS>
+{
+    protected _style: any;
+    protected _view: any;
+
+    constructor(options: BaseControllerOptions = {})
+    {
+        super();
+        this._style = options.style;
+        this._view = options.view;
+    }
+
+    public setView(): void
+    {
+        //
+    }
+
+    public setStyle(): void
+    {
+        //
```

Comment:
Useless function

Reply:
it is not implemented yet, and will not be in the scope of this PR, however it was causing issues with the react compiler that were preventing me from building/running the project. Commented out until implementation is completed.